lib/stat_tracker.rb

```diff
@@ -142,13 +142,31 @@ def worst_coach
   end
 
   def most_accurate_team
-
+    team_accuracies = calculate_team_accuracies
+    team_accuracies.max_by { |team, _| team_accuracies[team] }[0]
   end
 
   def least_accurate_team
+    team_accuracies = calculate_team_accuracies
+    team_accuracies.min_by { |team, _| team_accuracies[team] }[0]
+  end
+
+  def calculate_team_accuracies
+    team_goals = Hash.new { |hash, key| hash[key] = [0, 0] }
+    @game_teams.each do |game|
+      team_goals[game.team_id][0] += game.goals
+      team_goals[game.team_id][1] += game.shots
+    end
 
+    team_accuracies = {}
+    team_goals.each do |team_id, (goals, shots)|
+      accuracy = (goals.to_f / shots.to_f) * 100
+      team_name = team_name_by_id(team_id)
+      team_accuracies[team_name] = accuracy.round(2)
+    end
+
+    team_accuracies
```

Comment:
Looks like these tests are failing. Do you mind taking a look to make sure everything is operating as intended?

Reply:
Woops, sorry! These also need to take an argument of the season ID

Reply:
Be aware that the 'season' field in the CSV is a string of 8 chars. i.e. "20122013" ect.

Reply:
Good catch Weston!

Reply:
@topher-nullset Thanks for making those corrects! Everything is working as intended now with both our RSpec file and the spec harness! 🎉